server/methods/catalog.js

```diff
@@ -154,9 +154,15 @@ function createHandle(productHandle, productId) {
   }
 
   // we should check again if there are any new matches with DB
-  if (Products.find({
-    handle: handle
-  }).count() !== 0) {
+  // exception product._id needed for cases then double triggering happens
+  const newHandleCount = Products.find({
+    handle,
+    _id: {
+      $nin: [productId]
+    }
+  }).count();
```

Comment:
This fixed an infinite loop I was seeing, though I don't recall how to reproduce it. It should be obvious reading the code that the $nin should have been there.